### DISCUSSION TITLE
자바스크립트에서 문자열의 경우, 배열처럼 인덱스에 접근이 가능합니다. 가능한 이유가 무엇인가요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#48] [원시 값과 객체의 비교]
- 자바스크립트에서 문자열의 경우, 배열처럼 인덱스에 접근이 가능합니다. 가능한 이유가 무엇인가요?

**Comment by pipisebastian** [comment]
> 자바스크립트에서 문자열의 경우, 배열처럼 인덱스에 접근이 가능합니다. 가능한 이유가 무엇인가요?

문자열은 유사배열이기 때문에 배열처럼 인덱스를 통한 접근이나, .length와 같은 메서드를 사용가능합니다.
원시타입인 문자열을 배열처럼 접근하려고 할때, 자바스크립트 문자열을 임시로 객체로 만들어주어, 프로퍼티를 사용할 수 있게 해줍니다. 이때 만들어지는 객체를 래퍼객체라고 합니다.

**Comment by publdaze** [comment]
> - 자바스크립트에서 문자열의 경우, 배열처럼 인덱스에 접근이 가능합니다. 가능한 이유가 무엇인가요?
- 문자열은 배열처럼 인덱스를 통해 각 문자에 접근 가능한 유사 배열 객체이므로 인덱스에 접근이 가능함
